plugins/woocommerce/src/Internal/Integrations/WPConsentAPI.php

```diff
@@ -89,5 +97,16 @@ private function enqueue_consent_api_scripts() {
 			Constants::get_constant( 'WC_VERSION' ),
 			true
 		);
+
+		// Add data for the script above. `wp_enqueue_script` API does not allow data attributes,
+		// so we need a separate script tag and pollute the global scope.
+		wp_add_inline_script(
+			'wp-consent-api-integration',
+			sprintf(
+				'window.wc_order_attribution.params.consentCategory = %s;',
+				wp_json_encode( self::$CONSENT_CATEGORY )
+			),
+			'before'
+		);
```

Comment:
Preferably, I'd add a single data attribute to the `<script>` enqueued above, like:

```js
// plugins/woocommerce/src/Internal/Integrations/WPConsentAPI.php:90
wp_enqueue_script(
			'wp-consent-api-integration',
			plugins_url(
				"assets/js/frontend/wp-consent-api-integration{$this->get_script_suffix()}.js",
				WC_PLUGIN_FILE
			),
			array( 'wp-consent-api', 'wc-order-attribution' ),
			Constants::get_constant( 'WC_VERSION' ),
			true,
			array( 'data' => array( 'category' => 'marketing' ) )
		);
```
to produce 
```html
<script src="..." id="wp-consent-api-integration-js" data-category="marketing"></script>
```
which then would be used as
```js
// plugins/woocommerce/client/legacy/js/frontend/wp-consent-api-integration.js:8
document.currentString.dataset.category
```


But for some reason WP API limits that and forces us to use globals.